### PR TITLE
Add j/k keyboard shortcuts

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @maiamcc,

Please review the following commits I made in branch nicks/ch9619:

50328ffffe3e5c2e20fb13f8e5655b80c4404108 (2020-10-02 19:12:44 -0400)
Add j/k keyboard shortcuts

0ecb7ec011c38ec8163c8e12507caa1f2b84b843 (2020-10-02 18:33:55 -0400)
web: split SidebarItem into its own file

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/SidebarKeyboardShortcuts.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      case "j":
+      case "k":
+        // An array of sidebar items, plus one at the beginning for 'All'
+        let names = [""].concat(items.map(item => item.name))
+        let index = names.indexOf(selected)
+        let dir = e.key === "j" ? 1 : -1
+        let targetIndex = (index + dir + names.length) % names.length
+        let name = names[targetIndex]
+        let path = name ? pathBuilder.path(`/r/${name}`) : pathBuilder.path("/")
+        history.push(path)
```
Comment (maiamcc) [inline web/src/SidebarKeyboardShortcuts.tsx:49]: is this in use right now? (presumably it's to let you go "back" to the previously selected resource eventually, right?)

File: web/src/SidebarKeyboardShortcuts.test.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        pathBuilder={pathBuilder}
+      />
+    </MemoryRouter>
+  )
+  return root
+    .find(SidebarKeyboardShortcuts)
+    .instance() as SidebarKeyboardShortcuts
+}
+
+it("navigates forwards", () => {
```
Comment (maiamcc) [inline web/src/SidebarKeyboardShortcuts.test.tsx:28]: nice! worth a test that e.g. cmd+J/cmd+K don't have an effect? (i don't feel strongly)

**Review by maiamcc — APPROVED** [review]
🙌 ⌨️

**Comment by hyu** [comment]
I really just came here to say that being able to j/k is verrrry satisfying. ☺️

**Comment by nicks** [comment]
@hyu ya, TBH, I'm very surprised how big a dopamine hit this is. possibly because it's VERY fast, which makes Tilt feel faster / more responsive

**Comment by nicks** [comment]
i'm going to change the semantics so that it doesn't wrap, which i think is more of what i expect from other tools

File: web/src/SidebarKeyboardShortcuts.tsx (hunk 1 above)
Comment (nicks) [reply web/src/SidebarKeyboardShortcuts.tsx:49]: ya, this is how we do all navigations right now. (This is how `Link` element works under the hood, see https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/modules/Link.js#L103)

**Pull request merged**